### PR TITLE
[Fix] 'To Bill' status even if the amount is zero

### PR DESCRIPTION
**Pull request by Alchez**
Pull-Request

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
- [x] Have you lint your code locally prior to submission?
- [ ] Have you successfully run tests with your changes locally?
- [x] Does your commit message have an explanation for your changes and why you'd like us to include them?
- [ ] Docs have been added / updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Did you modify the existing test cases? If yes, why?

---

What type of a PR is this? 

- [ ] Changes to Existing Features
- [ ] New Feature Submissions
- [x] Bug Fix
- [ ] Breaking Change

--- 

- Motivation and Context (What existing problem does the pull request solve):

Due to an existing bug, Delivery Note documents that have a zero-amount show their status as "To Bill" even if invoices were made against them.

- Allowed the updating of billing status even if the Delivery Note's net total is zero.
- List view statuses have been updated to reflect zero-amount orders correctly.

---

- Related Issue(s):

#5006
#12433

---

- Screenshots (if applicable, remember, a picture tells a thousand words):
image

### CODE REVIEW

File: erpnext/controllers/status_updater.py [hunk 1]
```diff
@@ -281,16 +281,20 @@ def _update_percent_field(self, args, update_modified=True):
...
-						ifnull(sum(if(%(target_ref_field)s > %(target_field)s, abs(%(target_field)s), abs(%(target_ref_field)s))), 0)
-						/ sum(abs(%(target_ref_field)s)) * 100
-					from `tab%(target_dt)s` where parent="%(name)s" having sum(abs(%(target_ref_field)s)) > 0), 0), 6)
-					%(update_modified)s
-				where name='%(name)s'""" % args)
+			frappe.db.sql("""
+			UPDATE `tab%(target_parent_dt)s`
+			SET %(target_parent_field)s = round(ifnull(
+				(
+					SELECT IF(sum(abs(%(target_ref_field)s)) = 0, 100, ifnull(
```
Comment (nabinhait) [inline erpnext/controllers/status_updater.py]: Why setting `per_billed` as 100 directly, incase of zero grand total sales / purchase order? In those cases, we are already setting it based on `qty` using `update_billing_status_for_zero_amount_refdoc`?

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
@Alchez Any comments on the review?

**Comment by Alchez** [comment]
@nabinhait, sorry for the late reply.

I agree with your comment, but I can't think of a clean way to only update the DN ref field. Any suggestions on how to go about this?

**Comment by Alchez** [comment]
@nabinhait - I've removed the SQL query update, and made the fix that only applies to Delivery Note documents. Can you please check and review?

CC: @agritheory

**Comment by vjFaLk** [comment]
@nabinhait Sirji pls

**Comment by vjFaLk** [comment]
@rmehta Sirji pls

**Comment by neilLasrado** [comment]
@nabinhait @rmehta What is to be done here?

**Pull request merged**